### PR TITLE
Extend hoverxref_roles to custom crossrefs (#4495)

### LINKED ISSUE #4495: Extend hoverxref_roles
@humitos [suggested](https://github.com/scrapy/scrapy/issues/4475#issuecomment-613350667) extending the `hoverxref_roles` setting of the corresponding Sphinx extension so that the display-on-hover behavior of the documentations works for things like signal or setting references.

### PR DESCRIPTION
**Pull request by willbeaufoy**
Additionally, I think the documentation in docs/README.rst is out
of date: none of the commands built the docs properly for me (I had to
revert the changes in 901892da to docs/conf.py to get them to build
properly, and even then no tooltips displayed).

Building them with tox worked for me, but other developers say they
can still use the original method, so the docs now contain both.

Fixes #4495.

### CODE REVIEW

File: docs/README.rst [hunk 1]
```diff
@@ -4,56 +4,9 @@
...
-    pip install -r requirements.txt
-
-
-Compile the documentation
--------------------------
-
 To compile the documentation (to classic HTML output) run the following command
 from this dir::
 
-    make html
```
Comment (elacuesta) [inline docs/README.rst:27]: I don't know about this, it is in fact how I'm building the docs these days. `tox -e docs` gives me the following error, and I have stopped trying to understand how Sphinx works:

```
$ tox -e docs
GLOB sdist-make: /.../scrapy/setup.py
docs inst-nodeps: /.../scrapy/.tox/.tmp/package/1/Scrapy-2.1.0.zip
docs installed: alabaster==0.7.12,attrs==19.3.0,Automat==20.2.0,Babel==2.8.0,certifi==2020.4.5.1,cffi==1.14.0,chardet==3.0.4,constantly==15.1.0,cryptography==2.9.2,cssselect==1.1.0,docutils==0.16,hyperlink==19.0.0,idna==2.9,imagesize==1.2.0,incremental==17.5.0,Jinja2==2.11.2,lxml==4.5.0,MarkupSafe==1.1.1,packaging==20.3,parsel==1.6.0,Protego==0.1.16,pyasn1==0.4.8,pyasn1-modules==0.2.8,pycparser==2.20,PyDispatcher==2.0.5,Pygments==2.6.1,PyHamcrest==2.0.2,pyOpenSSL==19.1.0,pyparsing==2.4.7,pytz==2020.1,queuelib==1.5.0,requests==2.23.0,Scrapy==2.1.0,service-identity==18.1.0,six==1.14.0,snowballstemmer==2.0.0,Sphinx==3.0.3,sphinx-hoverxref==0.3b1,sphinx-notfound-page==0.4,sphinx-rtd-theme==0.4.3,sphinxcontrib-applehelp==1.0.2,sphinxcontrib-devhelp==1.0.2,sphinxcontrib-htmlhelp==1.0.3,sphinxcontrib-jsmath==1.0.1,sphinxcontrib-qthelp==1.0.3,sphinxcontrib-serializinghtml==1.1.4,Twisted==20.3.0,urllib3==1.25.9,w3lib==1.21.0,zope.interface==5.1.0
docs run-test-pre: PYTHONHASHSEED='2542448692'
docs run-test: commands[0] | sphinx-build -W -b html . /.../scrapy/.tox/docs/tmp/html
Running Sphinx v3.0.3
loading translations [en]... done
/.../scrapy/.tox/docs/lib/python3.6/site-packages/sphinx/deprecation.py:52: RemovedInSphinx40Warning: sphinx.builders.html.DirectoryHTMLBuilder is deprecated. Check CHANGES for Sphinx API modifications.
  return getattr(self._module, name)
making output directory... done
loading intersphinx inventory from https://coverage.readthedocs.io/en/stable/objects.inv...
loading intersphinx inventory from https://cssselect.readthedocs.io/en/latest/objects.inv...
loading intersphinx inventory from https://docs.pytest.org/en/latest/objects.inv...
loading intersphinx inventory from https://docs.python.org/3/objects.inv...
loading intersphinx inventory from https://www.sphinx-doc.org/en/master/objects.inv...
loading intersphinx inventory from https://tox.readthedocs.io/en/latest/objects.inv...
loading intersphinx inventory from https://twistedmatrix.com/documents/current/objects.inv...
loading intersphinx inventory from https://twistedmatrix.com/documents/current/api/objects.inv...
building [mo]: targets for 0 po files that are out of date
building [html]: targets for 80 source files that are out of date
updating environment: [new config] 80 added, 0 changed, 0 removed
reading sources... [  1%] README
reading sources... [  2%] contributing
reading sources... [  3%] faq
reading sources... [  5%] index
reading sources... [  6%] intro/examples
reading sources... [  7%] intro/install
reading sources... [  8%] intro/overview
reading sources... [ 10%] intro/tutorial
reading sources... [ 11%] news
reading sources... [ 12%] topics/api
reading sources... [ 13%] topics/architecture
reading sources... [ 15%] topics/asyncio
reading sources... [ 16%] topics/autothrottle
reading sources... [ 17%] topics/benchmarking
reading sources... [ 18%] topics/broad-crawls
reading sources... [ 20%] topics/commands
reading sources... [ 21%] topics/contracts
reading sources... [ 22%] topics/coroutines
reading sources... [ 23%] topics/debug
reading sources... [ 25%] topics/deploy
reading sources... [ 26%] topics/developer-tools
reading sources... [ 27%] topics/djangoitem
reading sources... [ 28%] topics/downloader-middleware
reading sources... [ 30%] topics/dynamic-content
reading sources... [ 31%] topics/email
reading sources... [ 32%] topics/exceptions
reading sources... [ 33%] topics/exporters
reading sources... [ 35%] topics/extensions
reading sources... [ 36%] topics/feed-exports
reading sources... [ 37%] topics/item-pipeline
reading sources... [ 38%] topics/items
reading sources... [ 40%] topics/jobs
reading sources... [ 41%] topics/leaks
reading sources... [ 42%] topics/link-extractors
reading sources... [ 43%] topics/loaders
reading sources... [ 45%] topics/logging
reading sources... [ 46%] topics/media-pipeline
reading sources... [ 47%] topics/practices
reading sources... [ 48%] topics/request-response
reading sources... [ 50%] topics/scrapyd
reading sources... [ 51%] topics/selectors
reading sources... [ 52%] topics/settings
reading sources... [ 53%] topics/shell
reading sources... [ 55%] topics/signals
reading sources... [ 56%] topics/spider-middleware
reading sources... [ 57%] topics/spiders
reading sources... [ 58%] topics/stats
reading sources... [ 60%] topics/telnetconsole
reading sources... [ 61%] topics/webservice
reading sources... [ 62%] venv-docs/lib/python3.7/site-packages/Jinja2-2.11.2.dist-info/LICENSE
reading sources... [ 63%] venv-docs/lib/python3.7/site-packages/alabaster-0.7.12.dist-info/DESCRIPTION
reading sources... [ 65%] venv-docs/lib/python3.7/site-packages/chardet-3.0.4.dist-info/DESCRIPTION
reading sources... [ 66%] venv-docs/lib/python3.7/site-packages/constantly-15.1.0.dist-info/DESCRIPTION
reading sources... [ 67%] venv-docs/lib/python3.7/site-packages/cryptography-2.9.2.dist-info/AUTHORS
reading sources... [ 68%] venv-docs/lib/python3.7/site-packages/hyperlink-19.0.0.dist-info/DESCRIPTION
reading sources... [ 70%] venv-docs/lib/python3.7/site-packages/idna-2.9.dist-info/LICENSE
reading sources... [ 71%] venv-docs/lib/python3.7/site-packages/imagesize-1.2.0.dist-info/LICENSE
reading sources... [ 72%] venv-docs/lib/python3.7/site-packages/incremental-17.5.0.dist-info/DESCRIPTION
reading sources... [ 73%] venv-docs/lib/python3.7/site-packages/pyasn1-0.4.8.dist-info/LICENSE
reading sources... [ 75%] venv-docs/lib/python3.7/site-packages/pytz-2020.1.dist-info/DESCRIPTION
reading sources... [ 76%] venv-docs/lib/python3.7/site-packages/queuelib-1.5.0.dist-info/DESCRIPTION
reading sources... [ 77%] venv-docs/lib/python3.7/site-packages/sphinx/ext/autosummary/templates/autosummary/base
reading sources... [ 78%] venv-docs/lib/python3.7/site-packages/sphinx/ext/autosummary/templates/autosummary/class
reading sources... [ 80%] venv-docs/lib/python3.7/site-packages/sphinx/ext/autosummary/templates/autosummary/module
reading sources... [ 81%] venv-docs/lib64/python3.7/site-packages/Jinja2-2.11.2.dist-info/LICENSE
reading sources... [ 82%] venv-docs/lib64/python3.7/site-packages/alabaster-0.7.12.dist-info/DESCRIPTION
reading sources... [ 83%] venv-docs/lib64/python3.7/site-packages/chardet-3.0.4.dist-info/DESCRIPTION
reading sources... [ 85%] venv-docs/lib64/python3.7/site-packages/constantly-15.1.0.dist-info/DESCRIPTION
reading sources... [ 86%] venv-docs/lib64/python3.7/site-packages/cryptography-2.9.2.dist-info/AUTHORS
reading sources... [ 87%] venv-docs/lib64/python3.7/site-packages/hyperlink-19.0.0.dist-info/DESCRIPTION
reading sources... [ 88%] venv-docs/lib64/python3.7/site-packages/idna-2.9.dist-info/LICENSE
reading sources... [ 90%] venv-docs/lib64/python3.7/site-packages/imagesize-1.2.0.dist-info/LICENSE
reading sources... [ 91%] venv-docs/lib64/python3.7/site-packages/incremental-17.5.0.dist-info/DESCRIPTION
reading sources... [ 92%] venv-docs/lib64/python3.7/site-packages/pyasn1-0.4.8.dist-info/LICENSE
reading sources... [ 93%] venv-docs/lib64/python3.7/site-packages/pytz-2020.1.dist-info/DESCRIPTION
reading sources... [ 95%] venv-docs/lib64/python3.7/site-packages/queuelib-1.5.0.dist-info/DESCRIPTION
reading sources... [ 96%] venv-docs/lib64/python3.7/site-packages/sphinx/ext/autosummary/templates/autosummary/base
reading sources... [ 97%] venv-docs/lib64/python3.7/site-packages/sphinx/ext/autosummary/templates/autosummary/class
reading sources... [ 98%] venv-docs/lib64/python3.7/site-packages/sphinx/ext/autosummary/templates/autosummary/module
reading sources... [100%] versioning


Warning, treated as error:
/.../scrapy/docs/venv-docs/lib/python3.7/site-packages/sphinx/ext/autosummary/templates/autosummary/base.rst:3:Error in "currentmodule" directive:
maximum 1 argument(s) allowed, 3 supplied.

.. currentmodule:: {{ module }}
ERROR: InvocationError for command /.../scrapy/.tox/docs/bin/sphinx-build -W -b html . /.../scrapy/.tox/docs/tmp/html (exited with code 2)
__________________________________________________________________________________________________________________ summary ___________________________________________________________________________________________________________________
ERROR:   docs: commands failed
```

Comment (elacuesta) [inline docs/README.rst:27]: I don't know about this, it's still working for me. If we're removing these docs we might as well update/remove the Makefile itself.

Comment (Gallaecio) [reply docs/README.rst:27]: I think just covering the Tox approach in the documentation makes sense, assuming there is no common use case where avoiding Tox usage is needed.

Someone interested in not using Tox should be able to figure things out from `tox.ini`.

Comment (willbeaufoy) [reply docs/README.rst:27]: @elacuesta interesting that `make html` still works for you, when I run it I get a warning saying "hoverxref extension is not fully configured. Tooltips may not work as expected." and indeed the tooltips don't work when viewing the generated docs.

Let me know what you guys want to do, I guess it's one of:

1) Only keep tox approach in the docs (I'll then remove the Makefile).
2) Keep both approaches in the docs.

Comment (kmike) [reply docs/README.rst:27]: It'd be nice to keep an option to use make, and all the make commands we have for docs, if possible.

Comment (willbeaufoy) [reply docs/README.rst]: @kmike done.

Comment (willbeaufoy) [reply docs/README.rst]: Is there anything more to do on this? As far as I know it's done.

File: docs/README.rst [hunk 2]
```diff
@@ -57,3 +57,12 @@ There is a way to recreate the doc automatically when you make changes, you
...
     make watch
+
+Alternative method using tox
+----------------------------
+
+To compile the documentation to HTML run the following command from this dir::
+
+    tox -e docs
+
+Documentation will be generated (in HTML format) inside the ``../.tox/docs/tmp/html`` dir.
```
Comment (elacuesta) [inline docs/README.rst:68]: It's not necessary to be inside the `docs` directory build the documentation with `tox`.

Comment (willbeaufoy) [reply docs/README.rst]: Oh yeah, thanks. Fixed now.

**Pull request merged**